### PR TITLE
hwmv2: Port efm32 socs and boards

### PR DESCRIPTION
**Pull request by nordicjm**

### CODE REVIEW

**Review by jerome-pouiller — COMMENTED** [review]
Hello Jamie,

Thank you to care take care about this. It seems the current classification of the Silabs boards is a bit messy. Maybe we could take advantage of HWMV2 to clean that?

I have extracted the relevant informations about the Silabs boards (mainly from [here][1]):
```
Current DTS name                            Board     Chip       SDID  Bundle
efr32_radio/efr32_radio_brd4104a.dts        BRD4104A  EFR32BG13P   89
efr32_radio/efr32_radio_brd4161a.dts        BRD4161A  EFR32MG12P   84
efr32_radio/efr32_radio_brd4170a.dts        BRD4170A  EFR32MG12P   84
efr32_radio/efr32_radio_brd4180a.dts        BRD4180A  EFR32MG21   200
efr32_radio/efr32_radio_brd4187c.dts        BRD4187C  EFR32MG24   215
efr32_radio/efr32_radio_brd4250b.dts        BRD4250B  EFR32FG1P    80
efr32_radio/efr32_radio_brd4255a.dts        BRD4255A  EFR32FG13P   89
efr32_thunderboard/efr32bg27_brd2602a.dts   BRD2602A  EFR32BG27   230
efr32_thunderboard/efr32bg22_brd4184a.dts   BRD4184A  EFR32BG22   205  "Thunderboard EFR32BG22" / SLTB010A (A00 to A01)
efr32_thunderboard/efr32bg22_brd4184b.dts   BRD4184B  EFR32BG22   205  "Thunderboard EFR32BG22" / SLTB010A (A02 to A04)
efr32mg_sltb004a/efr32mg_sltb004a.dts       BRD4166A  EFR32MG12P   84  "Thunderboard Sense 2"/ SLTB004A
efr32xg24_dk2601b/efr32xg24_dk2601b.dts     BRD2601B  EFR32MG24   215  "xG24 Dev Kit" / xG24-DK2601B
efm32wg_stk3800/efm32wg_stk3800.dts         BRD2400B  EFM32WG      74  "EFM32 Wonder Gecko Starter Kit"/ EFM32WG-STK3800
efm32gg_sltb009a/efm32gg_sltb009a.dts       BRD2207A  EFM32GG12B  106  "Thunderboard EFM32GG12" / SLTB009A
efm32gg_slwstk6121a/efm32gg_slwstk6121a.dts BRD4321A  WGM160      100  "WGM160P Wi-Fi Module Wireless Starter Kit" / SLWSTK6121A (module that include an EFR32GG11B)
efm32gg_stk3701a/efm32gg_stk3701a.dts       BRD2204C  EFM32GG11B  100  "EFM32GG11 Giant Gecko Starter Kit" / SLSTK3701A
efm32hg_slstk3400a/efm32hg_slstk3400a.dts   BRD2012A  EFM32HG      77  "EFM32 Happy Gecko Starter Kit" / SLSTK3400A
efm32pg_stk3401a/efm32pg_stk3401a.dts       BRD2500A  EFM32PG1B    80  "EFM32 Pearl Gecko Starter Kit" / SLSTK3401A
efm32pg_stk3402a/efm32pg_stk3402a.dts       BRD2501A  EFM32PG12B   84  "EFM32PG12 Pearl Gecko Starter Kit"
efm32pg_stk3402a/efm32pg_stk3402a_jg.dts    ??        EFM32JG12B   84
```

Do you think we should first merge this PR and then send a new PR to clean up the Silabs boards or we should include everything in one PR?
Do you still want to take care of that or it is out of the scope you want to address? If needed, maybe I could take over this PR. However, I have not so much time these days.

[1]: https://github.com/SiliconLabs/gecko_sdk/tree/gsdk_4.4/platform/Device/SiliconLabs

**Comment by nordicjm** [comment]
> Hello Jamie,
> 
> Thank you to care take care about this. It seems the current classification of the Silabs boards is a bit messy. Maybe we could take advantage of HWMV2 to clean that?
> 
> I have extracted the relevant informations about the Silabs boards (mainly from [here](https://github.com/SiliconLabs/gecko_sdk/tree/gsdk_4.4/platform/Device/SiliconLabs)):
> 
> ```
> Current DTS name                            Board     Chip       SDID  Bundle
> efr32_radio/efr32_radio_brd4104a.dts        BRD4104A  EFR32BG13P   89
> efr32_radio/efr32_radio_brd4161a.dts        BRD4161A  EFR32MG12P   84
> efr32_radio/efr32_radio_brd4170a.dts        BRD4170A  EFR32MG12P   84
> efr32_radio/efr32_radio_brd4180a.dts        BRD4180A  EFR32MG21   200
> efr32_radio/efr32_radio_brd4187c.dts        BRD4187C  EFR32MG24   215
> efr32_radio/efr32_radio_brd4250b.dts        BRD4250B  EFR32FG1P    80
> efr32_radio/efr32_radio_brd4255a.dts        BRD4255A  EFR32FG13P   89
> efr32_thunderboard/efr32bg27_brd2602a.dts   BRD2602A  EFR32BG27   230
> efr32_thunderboard/efr32bg22_brd4184a.dts   BRD4184A  EFR32BG22   205  "Thunderboard EFR32BG22" / SLTB010A (A00 to A01)
> efr32_thunderboard/efr32bg22_brd4184b.dts   BRD4184B  EFR32BG22   205  "Thunderboard EFR32BG22" / SLTB010A (A02 to A04)
> efr32mg_sltb004a/efr32mg_sltb004a.dts       BRD4166A  EFR32MG12P   84  "Thunderboard Sense 2"/ SLTB004A
> efr32xg24_dk2601b/efr32xg24_dk2601b.dts     BRD2601B  EFR32MG24   215  "xG24 Dev Kit" / xG24-DK2601B
> efm32wg_stk3800/efm32wg_stk3800.dts         BRD2400B  EFM32WG      74  "EFM32 Wonder Gecko Starter Kit"/ EFM32WG-STK3800
> efm32gg_sltb009a/efm32gg_sltb009a.dts       BRD2207A  EFM32GG12B  106  "Thunderboard EFM32GG12" / SLTB009A
> efm32gg_slwstk6121a/efm32gg_slwstk6121a.dts BRD4321A  WGM160      100  "WGM160P Wi-Fi Module Wireless Starter Kit" / SLWSTK6121A (module that include an EFR32GG11B)
> efm32gg_stk3701a/efm32gg_stk3701a.dts       BRD2204C  EFM32GG11B  100  "EFM32GG11 Giant Gecko Starter Kit" / SLSTK3701A
> efm32hg_slstk3400a/efm32hg_slstk3400a.dts   BRD2012A  EFM32HG      77  "EFM32 Happy Gecko Starter Kit" / SLSTK3400A
> efm32pg_stk3401a/efm32pg_stk3401a.dts       BRD2500A  EFM32PG1B    80  "EFM32 Pearl Gecko Starter Kit" / SLSTK3401A
> efm32pg_stk3402a/efm32pg_stk3402a.dts       BRD2501A  EFM32PG12B   84  "EFM32PG12 Pearl Gecko Starter Kit"
> efm32pg_stk3402a/efm32pg_stk3402a_jg.dts    ??        EFM32JG12B   84
> ```
> 
> Do you think we should first merge this PR and then send a new PR to clean up the Silabs boards or we should include everything in one PR? Do you still want to take care of that or it is out of the scope you want to address? If needed, maybe I could take over this PR. However, I have not so much time these days.

Things can be tided up after this is merged but should ideally be completed before the hwmv2 branch is ready for merging into `main`, when it is ready the whole branch will be squashed to a single commit of a few commits (which are going to be huge). If it's not urgent to be tired up for completion of hwmv2 then the changes can be submitted after (3.7 release is a long way away!). I'm fine if you want to take over this PR or create a new PR.

**Review by jerome-pouiller — COMMENTED** [review]
I made 3 comments I believe should be addressed in this PR. If you believe it too much work, we can accept this PR and I could send a new PR after.

Anyway, I will send a PR for other inconsistencies I have observed (especially, `brd2602a` should be a part of `efr32_thunderboard` and `CONFIG_BOARD` should have same value than `SILABS_GECKO_BOARD`).

File: boards/silabs/efr32_thunderboard/board.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+boards:
+  - name: efr32bg22_brd4184
+    vendor: silabs
+    revision:
+      format: letter
+      default: "A"
+      revisions:
+        - name: "A"
+        - name: "B"
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply boards/silabs/efr32_thunderboard/board.yml]: Fixed

File: drivers/hwinfo/Kconfig
[the remark this answers is not in the document]
Comment (nordicjm) [reply drivers/hwinfo/Kconfig:1]: You can send a commit splitting them up after

**Review by jerome-pouiller — APPROVED** [review]
I plan to send a PR on top this one to clean up boards names in the Silabs tree.

**Pull request merged**